accounting/accounting.go

```diff
@@ -97,6 +97,28 @@ func bytesAreZero(b []byte) bool {
 	return true
 }
 
+func (accounting *State) closeAccount(addr types.Address) {
+	if accounting.AlgoUpdates == nil {
+		accounting.AlgoUpdates = make(map[[32]byte]*idb.AlgoUpdate)
+	}
+
+	update, hasKey := accounting.AlgoUpdates[addr]
+
+	if !hasKey {
+		update = &idb.AlgoUpdate{
+			Balance: 0,
+			Rewards: 0,
+		}
+		accounting.AlgoUpdates[addr] = update
+		return
+	}
+
+	// If the key was there, override the rewards by setting to zero.
+	// The balance will be updated with the delta as usual.
+	update.Rewards = 0
```

Comment:
I don't think we should muck with this, I'm _pretty sure_ stxn.SenderRewards will be zero, but it could also be possible that `(previous balance - Fee + SenderRewards - Amount - ClosingAmount) == 0`

Reply:
This is required to handle the different edge cases.

Suppose 3 transactions evaluated in the following order
B closeTo A, closeToRewards = 100
A closes, [cumulative] rewards must be reset to 0
C closeTo A, closeToRewards = 100

At the end of all this A needs to have rewards = 100